applications/FluidDynamicsApplication/custom_processes/shock_detection_process.h

```diff
@@ -0,0 +1,366 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+#ifndef KRATOS_SHOCK_DETECTION_PROCESS
+#define KRATOS_SHOCK_DETECTION_PROCESS
+
+// System includes
+#include <string>
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "processes/process.h"
+#include "processes/calculate_nodal_area_process.h"
+#include "processes/compute_nodal_gradient_process.h"
+#include "processes/find_global_nodal_neighbours_process.h"
+#include "containers/global_pointers_vector.h"
+
+// Application includes
+#include "fluid_dynamics_application_variables.h"
+
+namespace Kratos
+{
+///@addtogroup FluidDynamicsApplication
+///@{
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @brief Auxiliary neighbour data class
+ * Auxiliary class to retrieve the data from the neighbours when communicating the pointers
+ * @tparam TShockVariableType Shock variable type
+ * @tparam TShockGradientVariableType Shock gradient variable type
+ */
+template<class TShockVariableType, class TShockGradientVariableType>
+class NeighbourData
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of NeighbourData
+    KRATOS_CLASS_POINTER_DEFINITION(NeighbourData);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief Construct a new Neighbour Data object
+     * Default neighbour data container constructor
+     * Required to compile the class
+     */
+    NeighbourData() = default;
+
+    /**
+     * @brief Construct a new Neighbour Data object
+     * Constructes a new neighbour data container instance
+     * @param rShockVariableValue Neighbour shock variable value
+     * @param rShockGradientVariableValue Neighbour shock variable gradient value
+     * @param rCoordinates Neighbour node coordinates
+     */
+    NeighbourData(
+        const typename TShockVariableType::Type& rShockVariableValue,
+        const typename TShockGradientVariableType::Type& rShockGradientVariableValue,
+        const array_1d<double, 3>& rCoordinates)
+    {
+        mCoordinates = rCoordinates;
+        mShockVariableValue = rShockVariableValue;
+        mShockGradientVariableValue = rShockGradientVariableValue;
+    }
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    array_1d<double, 3> mCoordinates;
+    typename TShockVariableType::Type mShockVariableValue;
+    typename TShockGradientVariableType::Type mShockGradientVariableValue;
+
+    ///@}
+private:
+    ///@name Serialization
+    ///@{
+
+    friend class Serializer;
+
+    void save(Serializer& rSerializer) const {}
```

Comment:
you need to provide implementation of those, otherwise the mpi serialization would fail

Reply:
Done.